build.rs

```diff
@@ -12,10 +12,27 @@
 // limitations under the License.
 
 use protobuf_build::Builder;
+use std::fs;
 
 fn main() {
+    let proto_dir = "proto";
+    let mut proto_files = Vec::new();
+
+    // Walk the proto directory and collect all .proto files.
+    for entry in fs::read_dir(proto_dir).unwrap() {
+        let entry = entry.unwrap();
+        let path = entry.path();
+        if path.is_file() {
+            if let Some(ext) = path.extension() {
+                if ext == "proto" {
+                    proto_files.push(path.to_string_lossy().into_owned());
+                }
+            }
+        }
+    }
+
     Builder::new()
-        .search_dir_for_protos("proto")
+        .files(&proto_files)
```

Comment:
Why remove `search_dir_for_protos`?

Reply:
`search_dir_for_protos` will include all files, the implement is not considered to exclude the no-proto files, it made the `make rust` failed.

and we add the OWNERS to this dir for best practice.

Reply:
ignore the no proto files in the commit:
https://github.com/pingcap/kvproto/pull/1231/commits/d9846b048e0156ec355a5e61252bd96ce6032920